commands.js

```diff
@@ -824,7 +824,7 @@ exports.commands = {
 			return this.errorReply("User '" + name + "' is unregistered, and so can't be promoted.");
 		}
 
-		let currentGroup = ((room.auth && room.auth[userid]) || (room.isPrivate !== true && Users.usergroups[userid]) || ' ').charAt(0);
+		let currentGroup = room.getAuth({userid, group: Users.usergroups[userid]}) || ' ';
```

Comment:
Considering Travis tests passed (there is no way they wouldn't try to load this file) and they are ran on Node.js 4.x, I would suspect it does work.